LB_D2Q9/pipe.py

```diff
@@ -250,6 +250,171 @@ def run(self, num_iterations):
             self.collide_particles() # Relax the nonequilibrium fields
 
 
+class Pipe_Flow_PeriodicBC(Pipe_Flow):
+
+    def __init__(self, **kwargs):
+        super(Pipe_Flow_PeriodicBC, self).__init__(**kwargs)
+
+    def move_bcs(self):
```

Comment:
A couple of thoughts here. You need to update the hydrodynamic variables in addition to imposing the boundary conditions on f. So, you need to specify rho and u at the boundaries in the init_hydro class, right?